agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -394,8 +397,14 @@ func (cs *CredentialSpecResource) handleS3CredentialspecFile(originalCredentials
 	}
 
 	resourceBase := filepath.Base(parsedARN.Resource)
-	localCredSpecFilePath := fmt.Sprintf("%s/s3_%s_%s.json", cs.credentialSpecResourceLocation, cs.taskARN, resourceBase)
+	taskArnSplit := strings.Split(cs.taskARN, "/")
+	length := len(taskArnSplit)
+	if length < 2 {
+		seelog.Errorf("Failed to retrieve taskId from taskArn.")
```

Comment:
nit: Why are you logging the same message twice?

Reply:
This comment is addressed.  For some reason my earlier response was not shown here. Posting again